docs/design/job_architecture.md

```diff
@@ -482,44 +306,187 @@ All cases:
+
+```Javascript
+define(
+  ['common/runtime', 'common/jobCommMessages'],
+  function(Runtime, jcm) {
+    const runtime = Runtime.make();
+
+    const listener = runtime.bus().listen({
+      channel: {
+        [jcm.CHANNEL.JOB]: 'some_job_id'
+      },
+      key: {
+        type: jcm.MESSAGE_TYPE.STATUS,
+      },
+      handle: (message) => {
+        ...process the message...
+      }
+    })
+  }
+);
+```
+
+The `on` function requires a constructed channel bus, premade and reusable for a given channel. So you would make a channel bus that would always receive messages for that channel, and instruct it on what to do when a message of a given type arrives. That looks like this:
+
+```Javascript
+define(
+  ['common/runtime', 'common/jobCommMessages'],
+  function(Runtime, jcm) {
+    const runtime = Runtime.make();
+
+const cellBus = runtime.bus().makeChannelBus({
+      name: {
+        cell: 'some_cell_id'
+      }
+    });
+    const listenerId = cellBus.on(jcm.MESSAGE_TYPE.RUN_STATUS, (message) => {
+      // process the message...
+    });
+  }
+);
+```
+
+Note that both of these create events that get bound to the DOM, and when the widget is removed, they should be cleaned up. This can be done by calling `bus.removeListener(id)` with the created `listenerId`. If you created a channel bus, then that bus should be used, otherwise the main `runtime.bus()` object should be used.
+
+
+## Job Management flow
+
+### JobManager initialization and startup.
 
-## JobManager initialization and startup.
 These steps take place whenever the user loads a narrative, or when the kernel is restarted. This ensures that the JobManager in the kernel is kept up-to-date on Job status.
-1. User starts kernel (opens a Narrative, or clicks Kernel -> Restart)
-2. `jobCommChannel.js` (front end widget) executes the following kernel call:
+
+* User starts kernel (opens a Narrative, or clicks Kernel -> Restart)
+* `jobCommChannel.js` (front end widget) executes the following kernel call:
 ```py
 JobManager().initialize_jobs()
-JobComm().start_update_loop
+JobComm().start_job_status_loop(cell_list=cell_list, init_jobs=True)
 ```
-3. `JobManager` does:
-  * Get current user and workspace id.
-  * `ExecutionEngine2.check_workspace_jobs` with the workspace id - gets the set of jobs in that workspace, and builds them into `Job` objects.
-4. `JobComm` does:
+* The `JobManager` runs `check_workspace_jobs` on EE2 with the current workspace ID to fetch the list of jobs associated with the workspace. This is filtered to include only those in `cell_list`, and then `Job` objects are built from the data.
+* `JobComm` does:
   * Starts the lookup loop thread.
   * On the first pass, this looks up status of all jobs and pushes them forward to the browser.

```

Comment:
Another wonderful feature is that if you reuse the same app cell for multiple job runs, all of those jobs are stored in the backend job manager -- there's currently no code that gets rid of old stored jobs when a cell is reset. I think at some point I had planned to tackle that (maybe as part of DATAUP-497) but since it wasn't critical path, I didn't.

Reply:
Well that's exciting, but out of scope

Reply:
:+1: